src/plugins/login/utils/simpleFileTokenCache.ts

```diff
@@ -0,0 +1,92 @@
+import fs from "fs";
+import path from "path";
+import os from "os";
+import * as adal from "adal-node";
+
+const CONFIG_DIRECTORY = path.join(os.homedir(), ".azure");
+const DEFAULT_SLS_TOKEN_FILE = path.join(CONFIG_DIRECTORY, "slsTokenCache.json");
+
+export class SimpleFileTokenCache implements adal.TokenCache {
+  private entries: any[] = [];
+  private subscriptions: any[] = [];
+
+  public constructor(private tokenPath: string = DEFAULT_SLS_TOKEN_FILE) {
+    this.load();
+  }
+
+  public add(entries: any, cb?: any) {
+    this.entries.push(...entries);
+    this.save();
+    if (cb) {
+      cb();
+    }
+  }
+
+  public remove(entries: any, cb?: any) {
+    this.entries = this.entries.filter(e => {
+      return !Object.keys(entries[0]).every(key => e[key] === entries[0][key]);
+    });
+    this.save();
+    if (cb) {
+      cb();
+    }
+  }
+
+  public find(query: any, cb?: any) {
+    let result = this.entries.filter(e => {
+      return Object.keys(query).every(key => e[key] === query[key]);
+    });
+    cb(null, result);
+    return result;
+  }
+
+  //-------- File toke cache specific methods
+
+  public addSubs(entries: any) {
+    this.subscriptions.push(...entries);
+    this.subscriptions = this.subscriptions.reduce((acc, current) => {
```

Comment:
what's `acc` stand for?

Reply:
it's the accumulator for the reducer, not a very exciting name haha